pipelines/tekton/tasks.go

```diff
@@ -0,0 +1,116 @@
+package tekton
+
+import (
+	pplnv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func taskFetchRepository() pplnv1beta1.PipelineTask {
+	return pplnv1beta1.PipelineTask{
+		Name: "fetch-repository",
+		TaskRef: &pplnv1beta1.TaskRef{
+			Name: "git-clone",
+		},
+		Workspaces: []pplnv1beta1.WorkspacePipelineTaskBinding{{
+			Name:      "output",
+			Workspace: "source-workspace",
+		}},
+		Params: []pplnv1beta1.Param{
+			{Name: "url", Value: *pplnv1beta1.NewArrayOrString("$(params.gitRepository)")},
+			{Name: "revision", Value: *pplnv1beta1.NewArrayOrString("$(params.gitRevision)")},
+		},
+	}
+}
+
+func taskkBuild(runAfter string) pplnv1beta1.PipelineTask {
+	return pplnv1beta1.PipelineTask{
+		Name: "build",
+		TaskRef: &pplnv1beta1.TaskRef{
+			Name: "buildpacks",
+		},
+		RunAfter: []string{runAfter},
+		Workspaces: []pplnv1beta1.WorkspacePipelineTaskBinding{
+			{
+				Name:      "source",
+				Workspace: "source-workspace",
+			},
+			{
+				Name:      "cache",
+				Workspace: "cache-workspace",
+			}},
+		Params: []pplnv1beta1.Param{
+			{Name: "APP_IMAGE", Value: *pplnv1beta1.NewArrayOrString("$(params.imageName)")},
+			{Name: "SOURCE_SUBPATH", Value: *pplnv1beta1.NewArrayOrString("$(params.contextDir)")},
+			{Name: "BUILDER_IMAGE", Value: *pplnv1beta1.NewArrayOrString("$(params.builderImage)")},
+		},
+	}
+}
+
+// TODO this should be part of the future func-build Tekton Task as a post-build step
+func taskImageDigest(runAfter string) pplnv1beta1.PipelineTask {
+	script := `#!/usr/bin/env bash
+               set -e
+
+			   func_file="/workspace/source/func.yaml"
+			   if [ "$(params.contextDir)" != "" ]; then
+			   		func_file = /workspace/source/$(params.contextDir)/func.yaml
```

Comment:
I think the spaces around the equal shouldn't be there.
Also it would be good tho have quotes around the value.

Reply:
```diff
diff --git a/pipelines/tekton/tasks.go b/pipelines/tekton/tasks.go
index 16783524..20efc1d8 100644
--- a/pipelines/tekton/tasks.go
+++ b/pipelines/tekton/tasks.go
@@ -49,18 +49,19 @@ func taskkBuild(runAfter string) pplnv1beta1.PipelineTask {
 // TODO this should be part of the future func-build Tekton Task as a post-build step
 func taskImageDigest(runAfter string) pplnv1beta1.PipelineTask {
        script := `#!/usr/bin/env bash
-               set -e
 
-                          func_file="/workspace/source/func.yaml"
-                          if [ "$(params.contextDir)" != "" ]; then
-                                       func_file = /workspace/source/$(params.contextDir)/func.yaml
-                          fi
+set -e
 
-                          sed -i "s|^image:.*$|image: $(params.image)|" $func_file
-                          echo "Function image name: $(params.image)"
+func_file="/workspace/source/func.yaml"
+if [ "$(params.contextDir)" != "" ]; then
+  func_file="/workspace/source/$(params.contextDir)/func.yaml"
+fi
 
-                          sed -i "s/^imageDigest:.*$/imageDigest: $(params.digest)/" $func_file
-                          echo "Function image digest: $(params.digest)"
+sed -i "s|^image:.*$|image: $(params.image)|" "$func_file"
+echo "Function image name: $(params.image)"
+
+sed -i "s/^imageDigest:.*$/imageDigest: $(params.digest)/" "$func_file"
+echo "Function image digest: $(params.digest)"
        `
 
        return pplnv1beta1.PipelineTask{
```

Reply:
possible improvement for future: using `yq` instead of `sed`.